presto-main/src/main/java/com/facebook/presto/sql/planner/plan/Patterns.java

```diff
@@ -187,6 +187,14 @@ private Patterns() {}
         }
     }
 
+    public static class Join
+    {
+        public static Property<JoinNode, JoinNode.Type> type()
```

Comment:
s/type/joinType

Reply:
I think the naming is actually correct. This matches the property name -- `com.facebook.presto.sql.planner.plan.JoinNode#getType`
See also `com.facebook.presto.sql.planner.plan.Patterns.TopN#step`